pandas/core/arrays/categorical.py

```diff
@@ -121,9 +121,10 @@ def func(self, other):
                 i = self.categories.get_loc(other)
                 ret = getattr(self._codes, opname)(i)
 
-                # check for NaN in self
-                mask = self._codes == -1
-                ret[mask] = False
+                if opname not in {"eq", "__eq__", "ge", "__ge__", "gt", "__gt__"}:
```

Comment:
this is very strange that you are checking for eq and __eq__ it’s just __eq__

Reply:
Yeah, changed.